wagtail/core/models.py

```diff
@@ -1585,16 +1585,28 @@ def get_url_parts(self, request=None):
             else:
                 site_id, root_path, root_url, language_code = possible_sites[0]
 
-        # If the active language code is a variant of the page's language, then
-        # use that instead
-        # This is used when LANGUAGES contain more languages than WAGTAIL_CONTENT_LANGUAGES
-        if get_supported_content_language_variant(translation.get_language()) == language_code:
-            language_code = translation.get_language()
+        use_wagtail_i18n = getattr(settings, 'WAGTAIL_I18N_ENABLED', False)
+
+        if use_wagtail_i18n:
+            # If the active language code is a variant of the page's language, then
+            # use that instead
+            # This is used when LANGUAGES contain more languages than WAGTAIL_CONTENT_LANGUAGES
+            try:
+                if get_supported_content_language_variant(translation.get_language()) == language_code:
+                    language_code = translation.get_language()
+            except LookupError:
+                # active language code is not a recognised content language, so leave
+                # page's language code unchanged
+                pass
```

Comment:
I find that `contextlib.suppress` better carry the intend (and unlike `except/pass`, it won't be picked up by static analysis tools as a potential error). For example:

```suggestion
            # If the active language code is a variant of the page's language, use that instead.
            # This is used when LANGUAGES contain more languages than WAGTAIL_CONTENT_LANGUAGES.
            # LookupError will be raised if the active language code is not a recognised content language
            # so leave the page's language code unchanged
            with contextlib.suppress(LookupError):
                if get_supported_content_language_variant(translation.get_language()) == language_code:
                    language_code = translation.get_language()
```

Reply:
@loicteixeira Can you give more details about the issue with static analysis tools? I'm not sure about the `contextlib.suppress` version being an improvement (it's an extra import, and I think that having both comments up-front makes the code less immediately understandable).

Reply:
Some tools might raise warnings when all there is to an `except` block is `pass` statement. Now typically it is reserved for bare excepts (i.e. `except:`) or too broad excepts (i.e. `except Exception:`) but some tools also check for any typed exceptions. [Bandit only checks for bare/board exception by default](https://docs.openstack.org/bandit/1.4.0/plugins/try_except_pass.html), so it wouldn't have picked up this one.

I don't think an extra import can be an issue really. I still believe `suppress` better express the intent from a code POV, but I can see your point of having two separate comments being more easily understandable.